src/v/config/configuration.cc

```diff
@@ -325,39 +335,47 @@ configuration::configuration()
   , partition_manager_shutdown_watchdog_timeout(
       *this,
       "partition_manager_shutdown_watchdog_timeout",
-      "A threshold value to detect partitions which shutdown might have been "
-      "stuck. After this threshold a watchdog in partition manager will log "
-      "information about partition shutdown not making progress",
+      "A threshold value to detect partitions which might have been stuck "
+      "while shutting down. After this threshold, a watchdog in partition "
+      "manager will log information about partition shutdown not making "
+      "progress.",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
       30s)
   , admin_api_require_auth(
       *this,
       "admin_api_require_auth",
-      "Whether admin API clients must provide HTTP Basic authentication "
-      "headers",
+      "Whether Admin API clients must provide HTTP basic authentication "
+      "headers.",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
       false)
   , seed_server_meta_topic_partitions(
       *this, "seed_server_meta_topic_partitions")
   , raft_heartbeat_interval_ms(
       *this,
       "raft_heartbeat_interval_ms",
-      "Milliseconds for raft leader heartbeats",
+      "Number of milliseconds for Raft leader heartbeats.",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
       std::chrono::milliseconds(150),
       {.min = std::chrono::milliseconds(1)})
   , raft_heartbeat_timeout_ms(
       *this,
       "raft_heartbeat_timeout_ms",
-      "raft heartbeat RPC timeout",
+      "Raft heartbeat RPC (remote procedure call) timeout. Raft uses a "
+      "heartbeat mechanism to maintain leadership authority and to trigger "
+      "leader elections. The `raft_heartbeat_interval_ms` is a periodic "
+      "heartbeat sent by the partition leader to all followers to assert its "
```

Comment:
The easy fix for the ducktape tests is to avoid the word "assert"

Reply:
should we reword to remove "assert" ?

Reply:
i think we should add exceptions for these cases in ducktape. we don't want to restrict the language we can use in descriptions.

Reply:
> i think we should add exceptions for these cases in ducktape. we don't want to restrict the language we can use in descriptions.

I agree, which is what I [attempted here](https://github.com/redpanda-data/redpanda/pull/23619/commits/d43b9596dff8b9ccad0402effdf2bfcbc493e5d3#diff-3f567de33923dec6fa4ca0462aa968416b58c956c824dd007cff23498c342785R127) - but the NodeCrash regex still needs changing. I suggested an easy workaround.

I don't think the description of the config really needs to be rendered to the log at startup, the value should suffice.